### ISSUE TITLE
How to do a clean re-install of plugin

### ISSUE THREAD

**Issue by timpritlove** [stcr/subscribe-to-comments-reloaded#205]
As my issue #104 is still not solved, I am trying to do a clean install of my plugin installation in my multisite setup.

I disabled the plugin system-wide and removed the `wp_subscribe_reloaded_subscribers` table and re-enabled the plugin. However, all the settings and the subscriptions are still there.

My questions:

* I expected to find per-blog tables storing STCR data but nothing like this exists in my installation. Why?
* Where are these settings stored?
* How can I completely reset the whole STCR data set and do a clean install?

**Comment by Reedyseth** [comment]
Hello @timpritlove I apologized for the late response.



> I expected to find per-blog tables storing STCR data but nothing like this exists in my installation. Why?

The creation of the table `wp_subscribe_reloaded_subscribers` uses the prefix of your current wordpress installation, on this case is `wp`. Check if you have another table created with another prefix, you can use :
```sql
SHOW TABLES;
```
> Where are these settings stored?

Beside the table  `wp_subscribe_reloaded_subscribers`, the StCR settings are store under the `wp_options` table and `wp_post` meta table with its correct table prefix. Use this to see them:
```
-- Get the subscriptions
SELECT * FROM wp_postmeta WHERE meta_key LIKE '%\_stcr@%';
-- Get the options
SELECT * FROM wp_options WHERE option_name LIKE 'subscribe_reloaded%';
-- Get the emails details
SELECT * FROM wp_subscribe_reloaded_subscribers;
```

> How can I completely reset the whole STCR data set and do a clean install?

Unistall the plugin withing the WordPress plugin area or manually by deleting the values on the 3 previous tables that I mentioned and then delete the plugin from the plugins folder.

Regards.!!

**Comment by timpritlove** [comment]
Thanks for the instructions. I will try reactivating the plugin after the cleanup.

I suggest you add a "cleanup" button to your settings for people who want to deactivate / start over with your plugin.

**Comment by Reedyseth** [comment]
This commit closes this request a89313b57a2ed00d6767b76efccfd52ed4fc3ea2. 

There is a new feature called **Safely Uninstall** that allow you to delete the plugin using the WordPress plugin interface. If you have the option set to **Yes** everything but the subscriptions created by the plugin will be wipeout. So after you made sure that you have this option to **Yes** you can deactivate the plugin and the delete it. Now you have to install the plugin via WordPress or Upload the plugin `zip` file and activate it, after this step all your settings will be as default and your subscriptions will remain.

I am going the leave this issue open until I add the support for multisite installations.

**Comment by Reedyseth** [comment]
> As my issue #104 is still not solved, I am trying to do a clean install of my plugin installation in my multisite setup.

I have [tested](https://github.com/stcr/subscribe-to-comments-reloaded/issues/104#issuecomment-242872938) the latest revision with your issue and now everything is working.
